packages/debugger/src/panels/breakpoints/pauseonexceptions.tsx

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import * as React from 'react';
+import {
+  MenuSvg,
+  ToolbarButton,
+  ToolbarButtonComponent
+} from '@jupyterlab/ui-components';
+import { IDebugger } from '../../tokens';
+import { Breakpoints } from './index';
+
+const PAUSE_ON_EXCEPTION_CLASS = 'jp-debugger-pauseOnExceptions';
+const PAUSE_ON_EXCEPTION_BUTTON_CLASS = 'jp-PauseOnExceptions';
+const PAUSE_ON_EXCEPTION_MENU_CLASS = 'jp-PauseOnExceptions-menu';
+
+export class PauseOnExceptionsWidget extends ToolbarButton {
+  constructor(props: PauseOnExceptions.IProps) {
+    super();
+    this.menu = new PauseOnExceptionsMenu({
+      service: props.service,
+      commands: {
+        registry: props.commands.registry,
+        pauseOnExceptions: props.commands.pauseOnExceptions
+      }
+    });
+
+    this.node.className = PAUSE_ON_EXCEPTION_CLASS;
+
+    this._props = props;
+    this._props.className = PAUSE_ON_EXCEPTION_BUTTON_CLASS;
+    this._props.service.eventMessage.connect((_, event): void => {
+      if (event.event === 'initialized' || event.event === 'terminated') {
+        this.onChange();
+      }
+    }, this);
+    this._props.enabled = this._props.service.pauseOnExceptionsIsValid();
+    this._props.service.pauseOnExceptionChanged.connect(this.onChange, this);
+  }
+
+  /**
+   * Called when the debugger is initialized or the filter changed.
+   */
+  onChange() {
+    const session = this._props.service.session;
+    const exceptionBreakpointFilters = session?.exceptionBreakpointFilters;
+    this._props.className = PAUSE_ON_EXCEPTION_BUTTON_CLASS;
+    if (this._props.service.session?.isStarted && exceptionBreakpointFilters) {
+      if (session.isPausingOnException()) {
+        this._props.className += ' lm-mod-toggled';
+      }
+      this._props.enabled = true;
+    } else {
+      this._props.enabled = false;
+    }
+    this.update();
+  }
+
+  /**
+   * open menu on click.
+   */
+  onclick = () => {
+    this.menu.open(
+      this.node.getBoundingClientRect().left,
+      this.node.getBoundingClientRect().bottom
+    );
+  };
+
+  render(): JSX.Element {
+    return <ToolbarButtonComponent {...this._props} onClick={this.onclick} />;
+  }
+
+  menu: PauseOnExceptionsMenu;
```

Comment:
Why do we need this menu to be public? Could it at least be readonly?

Reply:
No need, it's private since https://github.com/jupyterlab/jupyterlab/pull/13601/commits/f3074a505a53283f77116b407ada1b1806e2b08d